providers/dns/liquidweb/liquidweb.go

```diff
@@ -0,0 +1,141 @@
+)
+
+// Config is used to configure the creation of the DNSProvider
+type Config struct {
+	URL                string
+	Username           string
+	Password           string
+	Zone               string
+	Timeout            time.Duration
+	PollingInterval    time.Duration
+	PropagationTimeout time.Duration
+}
+
+// NewDefaultConfig returns a default configuration for the DNSProvider
+func NewDefaultConfig() *Config {
+	config := &Config{
+		Timeout:         env.GetOrDefaultSecond("LIQUID_WEB_TIMEOUT", 1*time.Minute),
+		PollingInterval: env.GetOrDefaultSecond("LIQUID_WEB_POLLING_INTERVAL", 2*time.Second),
+		PropagationTimeout: env.GetOrDefaultSecond("LIQUID_WEB_PROPAGATION_TIMEOUT",
+
+			2*time.Minute),
+	}
+
+	return config
+}
+
+// DNSProvider is an implementation of the acme.ChallengeProvider interface
+// that uses Liquid Web's REST API to manage TXT records for a domain.
+type DNSProvider struct {
+	config      *Config
+	recordIDs   map[string]int
+	recordIDsMu sync.Mutex
+	client      *lw.API
+}
+
+// NewDNSProvider returns a DNSProvider instance configured for Liquid Web.
+func NewDNSProvider() (*DNSProvider, error) {
+	config := NewDefaultConfig()
+	config.URL = env.GetOrDefaultString("LIQUID_WEB_URL", "")
+	config.Username = env.GetOrDefaultString("LIQUID_WEB_USERNAME", "")
+	config.Password = env.GetOrDefaultString("LIQUID_WEB_PASSWORD", "")
+	config.Zone = env.GetOrDefaultString("LIQUID_WEB_ZONE", "")
+
+	return NewDNSProviderConfig(config)
+}
+
+// NewDNSProviderConfig return a DNSProvider instance configured for Liquid Web.
+func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
+	if config == nil {
+		return nil, errors.New("liquidweb: the configuration of the DNS provider is nil")
+	}
+
+	if config.URL == "" {
+		return nil, fmt.Errorf("liquidweb: url is missing")
+	}
+
+	if config.Username == "" {
+		return nil, fmt.Errorf("liquidweb: username is missing")
+	}
+
+	if config.Password == "" {
+		return nil, fmt.Errorf("liquidweb: password is missing")
+	}
+
+	// Initial new LW go client.
+	client, err := lw.NewAPI(config.Username, config.Password, config.URL, int(config.Timeout.Seconds()))
+
+	if err != nil {
+		return nil, fmt.Errorf("liquidweb: could not create Liquid Web API client: %v", err)
+	}
+
+	return &DNSProvider{
+		config:    config,
+		recordIDs: make(map[string]int),
+		client:    client,
+	}, nil
+}
+
+// Timeout returns the timeout and interval to use when checking for DNS propagation.
+// Adjusting here to cope with spikes in propagation times.
+func (d *DNSProvider) Timeout() (time.Duration, time.Duration) {
+	return d.config.PropagationTimeout, d.config.PollingInterval
+}
+
+// Present creates a TXT record using the specified parameters
+func (d *DNSProvider) Present(domain, token, keyAuth string) error {
+	fqdn, value := dns01.GetRecord(domain, keyAuth)
+	params := &network.DNSRecordParams{
+		Name:  fqdn[0 : len(fqdn)-1],
+		RData: strconv.Quote(value),
+		Type:  "TXT",
+		Zone:  d.config.Zone,
```

Comment:
could you add a configurable TTL:
https://github.com/go-acme/lego/blob/e820634e0e43baff0ef61728cb8379f804913b72/providers/dns/auroradns/auroradns.go#L30